packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -247,18 +247,19 @@ export abstract class CredentialService {
     return this.credentialRepository.findById(connectionId)
   }
 
-  public async deleteById(credentialId: string, options?: DeleteCredentialOptions): Promise<void> {
-    const credentialRecord = await this.getById(credentialId)
+  public async deleteById(credentialRecordId: string, options?: DeleteCredentialOptions): Promise<void> {
+    const credentialRecord = await this.getById(credentialRecordId)
 
     await this.credentialRepository.delete(credentialRecord)
 
     if (options?.deleteAssociatedCredentials) {
       for (const credential of credentialRecord.credentials) {
         const formatService: CredentialFormatService = this.getFormatService(credential.credentialRecordType)
-        await formatService.deleteCredentialById(credentialRecord, options)
+        await formatService.deleteCredentialById(credentialRecordId)
```

Comment:
This is incorrect. We should be passing the credentialId / credentialRecordId, not the credential exchange id. (so `credential.credentialRecordId`). The indy credential service doesn't know about the `CredentialExchangeRecord.id`)